Library/Homebrew/env_config.rb

```diff
@@ -502,9 +502,6 @@ def automatically_set_no_install_from_api?
 
     sig { returns(T::Boolean) }
     def install_from_api?
-      return false if OS.unsupported_configuration?
-      return false unless Homebrew.default_prefix?
-
       ENV["HOMEBREW_NO_INSTALL_FROM_API"].blank?
```

Comment:
Given this we may want to consider replacing all `install_from_api?` calls to `!no_install_from_api?` and deleting this